src/ast/passes/resource_analyser.cpp

```diff
@@ -132,6 +132,21 @@ void ResourceAnalyser::visit(Call &call)
     resources_.join_args.push_back(delim);
     resources_.needs_join_map = true;
   }
+  else if (call.func == "hist")
+  {
+    auto &r = resources_.hist_bits_arg;
+    int old, bits = static_cast<Integer *>(call.vargs->at(1))->n;
+    if (r.find(call.map->ident) != r.end() &&
+        (old = r[call.map->ident]) != bits)
+    {
+      LOG(ERROR, call.loc, err_)
+          << "Different bits in a single hist, had " << old << " now " << bits;
```

Comment:
```suggestion
    int bits = static_cast<Integer *>(call.vargs->at(1))->n;
    if (r.find(call.map->ident) != r.end() &&
        (r[call.map->ident]) != bits)
    {
      LOG(ERROR, call.loc, err_)
          << "Different bits in a single hist, had " << r[call.map->ident] << " now " << bits;
```
The multiple declarations and assignment/comparision in single lines feel too clever for me. Since this is just an error path there's no need to optimise it anyway, so let's just read the map twice to simplify the code.